presto-main/src/test/java/com/facebook/presto/operator/TestingBlockBuilders.java

```diff
@@ -0,0 +1,282 @@
+    private static final int MAX_STRING_SIZE = 10;
+
+    private static final Random RANDOM = new Random(0);
+
+    Block buildNullBlock(Type type, int positionCount)
+    {
+        return new RunLengthEncodedBlock(type.createBlockBuilder(null, 1).appendNull().build(), positionCount);
+    }
+
+    Block buildBigintBlock(int positionCount, boolean allowNulls)
+    {
+        return createLongsBlock(
+                IntStream.range(0, positionCount)
+                        .mapToObj(i -> allowNulls && i % 7 == 0 ? null : RANDOM.nextLong())
+                        .collect(Collectors.toList()));
+    }
+
+    Block buildLongDecimalBlock(int positionCount, boolean allowNulls)
+    {
+        return createLongDecimalsBlock(
+                IntStream.range(0, positionCount)
+                        .mapToObj(i -> allowNulls && i % 7 == 0 ? null : Double.toString(RANDOM.nextDouble() * RANDOM.nextInt()))
+                        .collect(Collectors.toList()));
+    }
+
+    Block buildIntegerBlock(int positionCount, boolean allowNulls)
+    {
+        return createIntsBlock(
+                IntStream.range(0, positionCount)
+                        .mapToObj(i -> allowNulls && i % 7 == 0 ? null : RANDOM.nextInt())
+                        .collect(Collectors.toList()));
+    }
+
+    Block buildSmallintBlock(int positionCount, boolean allowNulls)
+    {
+        return createTypedLongsBlock(
+                SMALLINT,
+                IntStream.range(0, positionCount)
+                        .mapToObj(i -> allowNulls && i % 7 == 0 ? null : RANDOM.nextLong() % Short.MIN_VALUE)
+                        .collect(Collectors.toList()));
+    }
+
+    Block buildBooleanBlock(int positionCount, boolean allowNulls)
+    {
+        return createBooleansBlock(
+                IntStream.range(0, positionCount)
+                        .mapToObj(i -> allowNulls && i % 7 == 0 ? null : RANDOM.nextBoolean())
+                        .collect(Collectors.toList()));
+    }
+
+    Block buildVarcharBlock(int positionCount, boolean allowNulls, int maxStringLength)
+    {
+        return createStringsBlock(
+                IntStream.range(0, positionCount)
+                        .mapToObj(i -> allowNulls && i % 7 == 0 ? null : getRandomString(maxStringLength))
+                        .collect(Collectors.toList()));
+    }
+
+    DictionaryBlock buildDictionaryBlock(Block dictionary, int positionCount)
+    {
+        int[] ids = IntStream.range(0, positionCount).map(i -> RANDOM.nextInt(dictionary.getPositionCount() / 10)).toArray();
+        return new DictionaryBlock(dictionary, ids);
+    }
+
+    RunLengthEncodedBlock buildRleBlock(Block block, int positionCount)
+    {
+        checkArgument(block.getPositionCount() > 0);
+        return new RunLengthEncodedBlock(block.getRegion(block.getPositionCount() / 2, 1), positionCount);
+    }
+
+    Block buildDictRleBlock(Block block, int positionCount, Optional<String> wrappings)
```

Comment:
- `wrappings` seems to describe a sequence of dictionary and RLE wrapping over a block; given that there are only few types of wrapping (e.g. Dictionary and RLE), I'd expect the sequence of these to be modeled as a list of enums
- any reason not to use enums here?
- even without enums, I'd use `switch` statement combined with `toCharArray` like below
- avoid mutating input parameters (block here); doing so hurts readability

```
    static Block buildDictRleBlock(Block block, int positionCount, Optional<String> wrappings)
    {
        if (!wrappings.isPresent()) {
            return null;
        }
        
        Block wrappedBlock = block;
        
        char[] codes = wrappings.get().toCharArray();
        for (int i = codes.length - 1; i >= 0; i--) {
            switch (codes[i]) {
                case 'D':
                    wrappedBlock = buildDictionaryBlock(wrappedBlock, positionCount);
                    break;
                case 'R':
                    wrappedBlock = buildRleBlock(wrappedBlock, positionCount);
                    break;
                default:
                    throw new IllegalArgumentException(format("wrappings %s is incorrect", wrappings));
            }
        }
        return wrappedBlock;
    }
```